### DISCUSSION TITLE
Expected UTM emulation performance with secure boot and disk encryption enabled

### DISCUSSION THREAD

**Discussion by subesokun** [utmapp/UTM#5522] [UTM Help (macOS)] [answered]
Hi,

I'm a bit wondering what the expected UTM emulation performance is when secure boot and disk encryption is enabled. I do know that emulation comes with certain costs but I'm uncertain if something is broken with my setup or if that is as good as it can get on a M1 Pro. Each UI interaction has a delay of 1-3 seconds which makes working with it almost impossible. For some reason the `QEMULauncher` process never exceeds +104 % CPU while interacting with the UI (only during boot time).

[Video showing the UI in action](https://github.com/utmapp/UTM/assets/6920421/814d0f64-70f2-4de2-80df-a2b81d7a7ee7).

Host:
MacBook M1 Pro (16GB RAM)

UTM System:
Arch: x86_64
System: Standard PC (Q35 + ICH9, 2009)
RAM: 8GB
CPU: Default
CPU Cores: 4

UTM QEMU:
UEFI Boot: y
RNG Device: y
TPM 2.0 Device: y
User local time for base clock: y

UTM Display:
Card: virtio-gpu-pci

Guest:
Freshly installed Ubuntu 22.04.2 AMD64 Desktop
Secure Boot enabled
LVM disk encryption enabled

**Comment by osy** [comment]
Disk encryption for VMs is not recommended. It will use the CPU to encrypt (instead of dedicated hardware like in macOS). If you wish to protect your files, use FileVault on your Mac to encrypt all files. If you want to encrypt it for transferring to another computer, use a dedicated tool such as an encrypted disk image with Disk Utility. There is zero security benefit of having it encrypted in the guest OS and TPM adds no security at all.

**Reply by subesokun** [reply]
Thanks for your feedback! I'd like to prevent enabling disk encryption for your mentioned reasons. However, to enroll my device into our IT some security policies are getting enforced like secure boot and disk encryption. I guess then this is as fast as it can get with emulation, or?

**Reply by osy** [reply, marked as answer]
That is unfortunate but it would seem so. In theory a lot of work can be put into paravirtualizing the block IO to handle encryption on the host side but it would be enormous amount of work and unlikely to benefit most people...

**Comment by subesokun** [comment]
I was spinning up a new VM in emulation mode (Ubuntu 22.04) in its default config and didn't enable any encryption. But UI is still very unresponsive and behaves like the VM where secure boot and disk encryption is enabled. Really wondering what I'm doing wrong here... For others the performance of an emulated Ubuntu Desktop is fine, right? 
Update: Also tried now a ARM64 Ubuntu Desktop and there the performance it very good. However, I'll need the x86_64 one due to some missing tools on Ubuntu ARM :(
